### PR TITLE
Adjust swagger-ui path if HTTP root is set

### LINKED ISSUE #5818: with quarkus.http.root-path set to abc the swagger-ui is not available
**Describe the bug**

with quarkus.http.root-path set to abc the swagger-ui is not available

**Expected behavior**
showing the sawgger-ui by http://localhost:8080/abc/swagger-ui 

**Actual behavior**
failing with JavaScript asset not loaded exception in the browser.

**To Reproduce**
1. in application.propertes - set quarkus.http.root-path to abc
2. go by http://localhost:8080/abc/swagger-ui  in the browser
3. see the exception in the browser console

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Fixes #5818

### CODE REVIEW

File: extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java [hunk 1]
```diff
@@ -118,7 +118,8 @@ public void registerSwaggerUiServletExtension(SwaggerUiRecorder recorder,
                     throw new RuntimeException(e);
                 }
             }
-            Handler<RoutingContext> handler = recorder.handler(cached.cachedDirectory, swaggerUiConfig.path);
+            Handler<RoutingContext> handler = recorder.handler(cached.cachedDirectory,
+                    httpRootPathBuildItem.adjustPath(swaggerUiConfig.path));
             routes.produce(new RouteBuildItem(swaggerUiConfig.path, handler));
             routes.produce(new RouteBuildItem(swaggerUiConfig.path + "/*", handler));
```
Comment (gsmet) [inline extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java]: The routes don't need the adjusted path?

File: extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java [hunk 2]
```diff
@@ -118,7 +118,8 @@ public void registerSwaggerUiServletExtension(SwaggerUiRecorder recorder,
...
             displayableEndpoints.produce(new NotFoundPageDisplayableEndpointBuildItem(swaggerUiConfig.path + "/"));
```
Comment (gsmet) [inline extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java]: I think at least this one needs it.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java (hunk 1 above)
Comment (stuartwdouglas) [reply extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java]: It is handled automatically, as the underlying router is mounted at the http root.

File: extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java (hunk 2 above)
Comment (stuartwdouglas) [reply extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java]: This gets adjusted in the processor that handles the routes, but turns out none of the others do.

Comment (stuartwdouglas) [reply extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java]: I am going to open a new PR for the fixing the not found page, as it is not really related.

**Comment by stuartwdouglas** [comment]
@gsmet I have opened a new PR to address the existing problems with the not found page

**Review by gsmet — APPROVED** [review]

**Pull request merged**